src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const result = [];
+
+  actions.forEach((action) => {
+    newState = { ...newState };
+
+    if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((elem) => delete newState[elem]);
+    } else if (action.type === 'clear') {
+      newState = {};
+    }
```

Comment:
better to rewrite it using the switch case

Reply:
ok, already fixed